src/hotspot/share/opto/buildOopMap.cpp

```diff
@@ -466,8 +466,8 @@ static void do_liveness(PhaseRegAlloc* regalloc, PhaseCFG* cfg, Block_List* work
           assert(def != 0, "input edge required");
           int first = regalloc->get_reg_first(def);
           int second = regalloc->get_reg_second(def);
-          if( OptoReg::is_valid(first) ) set_live_bit(tmp_live,first);
-          if( OptoReg::is_valid(second) ) set_live_bit(tmp_live,second);
+          if( OptoReg::is_valid(first) && !def->get_removed()) set_live_bit(tmp_live,first);
+          if( OptoReg::is_valid(second) && !def->get_removed()) set_live_bit(tmp_live,second);
```

Comment:
You need `def->as_Mach()->get_removed()` because `def ` is `Node*`.

Reply:
Done